sql/scan.go

```diff
@@ -130,8 +130,10 @@ func (n *scanNode) Next() bool {
 
 			// This is the first key for the row, initialize the column values that
 			// are part of the primary key.
-			for id := range n.qvals {
-				n.qvals[id].Expr = vals[id]
+			for id, val := range vals {
+				if qval := n.qvals[id]; qval != nil {
```

Comment:
if you write `qval, ok` then passing reads may wonder if `nil` is a possible value for `qval`. Why overcomplicate?